src/lighteval/models/transformers/transformers_model.py

```diff
@@ -724,176 +703,161 @@ def loglikelihood(
-                max_cont_tokens_length = max(len(c[0]) for c in batch_cont_tokens)
-                # These are the true lengths of the continuation tokens, we have to save them to be able to removed padding tokens from the generated tokens.
-                batch_cont_token_lengths = torch.tensor([c.shape[1] for c in batch_cont_tokens], device=self.device)
-                batch_cont_tokens = torch.cat(
-                    [
-                        F.pad(c, (0, max_cont_tokens_length - c.shape[1], 0, 0), value=self.tokenizer.pad_token_id)
-                        for c in batch_cont_tokens
-                    ],
-                    dim=0,
-                )
-                batch_cont_tokens, _ = self.pad_and_gather(batch_cont_tokens)
-                # Can be gathered as such
-                logits = torch.tensor(logits_sum, device=self.device)
-                max_equal = torch.tensor(max_equals, device=self.device)
-                batch_truncated = torch.tensor(prepared_batch.truncated, device=self.device)
-                batch_padded = torch.tensor(prepared_batch.padded, device=self.device)
-                if self.accelerator:
-                    logits = self.accelerator.gather_for_metrics(logits)
-                    max_equal = self.accelerator.gather_for_metrics(max_equal)
-                    batch_truncated = self.accelerator.gather_for_metrics(batch_truncated)
-                    batch_padded = self.accelerator.gather_for_metrics(batch_padded)
-                    batch_cont_token_lengths = self.accelerator.gather_for_metrics(batch_cont_token_lengths)
-
-                for logit, cont_tokens, maxe, batched_input, trunc, padded, len_input, len_token in zip(
-                    logits,
-                    batch_cont_tokens,
-                    max_equal,
-                    batched_inputs,
-                    batch_truncated,
-                    batch_padded,
-                    len_inputs,
-                    batch_cont_token_lengths,
-                ):
-                    # Filter out padding tokens from input_tokens and generated_tokens
-                    input_tokens = batched_input[:len_input].cpu().tolist()
-                    generated_tokens = cont_tokens[:len_token].cpu().tolist()
-
-                    answer = LoglikelihoodResponse(
-                        # todo: we might want to store the logits unsummed
-                        result=(float(logit.sum()), bool(maxe)) if return_bool_score else float(logit.sum()),
-                        input_tokens=input_tokens,
-                        generated_tokens=generated_tokens,
-                        truncated_tokens_count=trunc.cpu().item(),
-                        padded_tokens_count=padded.cpu().item(),
+                logits = F.log_softmax(model_output, dim=-1)  # [batch, sequence_length, vocab]
+
+                flat_index = 0
+                for i, doc in enumerate(batch):
+                    # Get the size of the corresponding nested list
```

Comment:
- could be worth adding a small explanation on the fact that you are "unflattening" the logits because of multichoice
- are we sure sorting of logits is correct even with parallelism?